providers/curation/mongoCurationStore.js

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation and others. Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+const MongoClient = require('mongodb').MongoClient
+const promiseRetry = require('promise-retry')
+const EntityCoordinates = require('../../lib/entityCoordinates')
+const throat = require('throat')
+const Curation = require('../../lib/curation')
+
+class MongoCurationStore {
+  constructor(options) {
+    this.options = options
+  }
+
+  initialize() {
+    return promiseRetry(async retry => {
+      try {
+        this.client = await MongoClient.connect(
+          this.options.connectionString,
+          { useNewUrlParser: true }
+        )
+        this.db = this.client.db(this.options.dbName)
+        this.collection = this.db.collection('curations')
+      } catch (error) {
+        retry(error)
+      }
+    })
+  }
+
+  /**
+   * Store the given set of curations overwriting any previous content. This effectively replicates the
+   * content of the files in the GitHub curation repo.
+   * @param {[Curation]} curations
+   */
+  async updateCurations(curations) {
+    await Promise.all(
+      curations.map(
+        throat(10, async curation => {
+          const _id = this._getCurationId(curation.data.coordinates)
+          await this.collection.replaceOne({ _id }, { _id, ...curation.data }, { upsert: true })
+        })
+      )
+    )
+    return null
+  }
+
+  /**
+   * Update the contribution entry for the given PR and record the associated curations -- the
+   * actual file content in the PR. If the curations are not provided, just the PR data is updated,
+   * any existing curation data is preserved.
+   * @param {*} pr - the PR object as provided by GitHub
+   * @param {[Curation]} curations? - The set of actual proposed changes.
+   */
+  updateContribution(pr, curations = null) {
+    if (curations) {
+      const files = {}
+      if (curations) curations.forEach(curation => (files[curation.path] = curation.data))
+      return this.collection.replaceOne({ _id: pr.number }, { _id: pr.number, pr, files }, { upsert: true })
+    }
+    // TODO reconsider `upsert` here. Great for resiliency but will it result in undetected inconsistent data?
+    return this.collection.updateOne({ _id: pr.number }, { $set: { pr: pr } }, { upsert: true })
```

Comment:
not sure we need `curations` to be optional here. We pull the set of files from GH based on PR number and commit ref from head of PR. 

The only way there are no curations is if the PR is empty. In this case, we shouldn't need to store anything and this could just always do a full replace above

Is there another case where curations is null?